scripts/zones/Kuftal_Tunnel/mobs/Guivre.lua

```diff
@@ -6,310 +6,225 @@ require("scripts/globals/pathfind")
+        xi.path.pathToNearest(mob, pathNodesB)
+    end
+end
+
+entity.onMobInitialize = function(mob)
+    mob:setMod(xi.mod.MOVE, 150)
+    mob:setMobMod(xi.mobMod.SIGHT_RANGE, 30)
+    mob:setMod(xi.mod.DOUBLE_ATTACK, 25)
+    mob:setMod(xi.mod.TRIPLE_ATTACK, 15)
 end
 
 entity.onMobSpawn = function(mob)
-    entity.onPath(mob)
+    --Let's set a pauseTime for when/if mob pauses at a point.
+    --Next, set a path to memory as well as if that path is to be
+    --traversed in reverse or not.
+    --Path choice can be A, or B.
+    --1: Path A
+    --2: Path B
+    mob:setLocalVar("pauseTime", math.random(8,20) + os.time())
+    mob:setLocalVar("mobPath", math.random(1,2))
+    mob:setLocalVar("pathReverse", math.random(0,1))
+    --onMobRoam will fire next.
 end
 
 entity.onMobRoam = function(mob)
-    -- move to start position if not moving
-    if not mob:isFollowingPath() then
-        xi.path.pathToNearest(mob, pathNodes)
+    -- First, let's check if mob is at spawn and needs a new path.
+    local nextIndex = mob:getLocalVar("nextPatrolIndex")
+    local distance = mob:checkDistance(102, -0.191, 3)
+    if nextIndex >= 24 and distance < 3 then
+        mob:setLocalVar("nextPatrolIndex", 2)
+        entity.onMobSpawn(mob)
+        return
+    end
+    --Next, let's check if mob is currently set to pause.
+    local pauseCheck = mob:getLocalVar("pauseTime")
+    if pauseCheck > os.time() then
+        --Mob is currently set to pause and not move for the amount of pauseTime
+        mob:setMobMod(xi.mobMod.NO_MOVE, 1)
+        --Guivre rotates while paused
+        local angleCheck = math.random(0,2)
+        if angleCheck == 0 then
+            mob:setRotation(math.random(45, 180))
+        elseif angleCheck == 1 then
+            mob:setRotation(math.random(135, 180))
+        elseif angleCheck == 2 then
+            mob:setRotation(180)
+        end
+        return
+    else
+        --Mob is not set to pause.
+        mob:setMobMod(xi.mobMod.NO_MOVE, 0)
+        local currentPath = mob:getLocalVar("mobPath")
+        local reverseCheck = mob:getLocalVar("pathReverse")
+        if currentPath == 1 and reverseCheck == 0 then
+            xi.path.patrol(mob, pathNodesA)
+            return
+        elseif currentPath == 2 and reverseCheck == 0 then
+            xi.path.patrol(mob, pathNodesB)
+            return
+        elseif currentPath == 1 and reverseCheck == 1 then
+            xi.path.patrol(mob,xi.path.fromEnd(pathNodesA, 1))
+            return
+        elseif currentPath == 2 and reverseCheck == 1 then
+            xi.path.patrol(mob,xi.path.fromEnd(pathNodesB, 1))
+            return
+        end
+    end
+    --Next, let's check if a mob is currently on a path.
+    if mob:isFollowingPath() then
+        xi.path.pathToNearest(mob, pathNodesB)
+        entity.onPath(mob)
+    else
+        --if not on pause or on a path, let's check if around path starting point
+        local distance = mob:checkDistance(102, -0.191, 3)
```

Comment:
This line is throwing an error and actually isn't needed because its declared at the start of the onMobRoam function.

Reply:
Got it--will update!